lib/plugins/aws/package/compile/events/cognitoUserPool/index.test.js

```diff
@@ -368,6 +368,219 @@ describe('AwsCompileCognitoUserPoolEvents', () => {
+              {
+                Trigger: 'CustomMessage',
+              },
+            ],
+          },
+        });
+      });
+    });
+
+    it('should create the necessary resources for a service using multiple event definitions', () => {
+      awsCompileCognitoUserPoolEvents.serverless.service.functions = {
+        first: {
+          name: 'first',
+          events: [
+            {
+              cognitoUserPool: {
+                pool: 'existing-cognito-user-pool',
+                trigger: 'CustomMessage',
+                existing: true,
+              },
+            },
+            {
+              cognitoUserPool: {
+                pool: 'existing-cognito-user-pool',
+                trigger: 'PreSignUp',
+                existing: true,
+              },
+            },
+            {
+              cognitoUserPool: {
+                pool: 'existing-cognito-user-pool',
+                trigger: 'DefineAuthChallenge',
+                existing: true,
+              },
+            },
+          ],
+        },
+      };
+
+      return expect(
+        awsCompileCognitoUserPoolEvents.existingCognitoUserPools()
+      ).to.be.fulfilled.then(() => {
+        const {
+          Resources,
+        } = awsCompileCognitoUserPoolEvents.serverless.service.provider.compiledCloudFormationTemplate;
+
+        expect(addCustomResourceToServiceStub).to.have.been.calledOnce;
+        expect(addCustomResourceToServiceStub.args[0][1]).to.equal('cognitoUserPool');
+        expect(addCustomResourceToServiceStub.args[0][2]).to.deep.equal([
+          {
+            Action: [
+              'cognito-idp:ListUserPools',
+              'cognito-idp:DescribeUserPool',
+              'cognito-idp:UpdateUserPool',
+            ],
+            Effect: 'Allow',
+            Resource: '*',
+          },
+          {
+            Action: ['lambda:AddPermission', 'lambda:RemovePermission'],
+            Effect: 'Allow',
+            Resource: 'arn:aws:lambda:*:*:function:first',
+          },
+        ]);
+        expect(Resources.FirstCustomCognitoUserPool1).to.deep.equal({
+          Type: 'Custom::CognitoUserPool',
+          Version: 1,
+          DependsOn: ['FirstLambdaFunction', 'CustomDashresourceDashexistingDashcupLambdaFunction'],
+          Properties: {
+            ServiceToken: {
+              'Fn::GetAtt': ['CustomDashresourceDashexistingDashcupLambdaFunction', 'Arn'],
+            },
+            FunctionName: 'first',
+            UserPoolName: 'existing-cognito-user-pool',
+            UserPoolConfigs: [
+              {
+                Trigger: 'CustomMessage',
+              },
+              {
+                Trigger: 'PreSignUp',
+              },
+              {
+                Trigger: 'DefineAuthChallenge',
+              },
+            ],
+          },
+        });
+      });
+    });
+
+    it('should create DependsOn clauses when one cognito user pool is used in more than 1 custom resources', () => {
```

Comment:
We may also add tests, where user configures two different pools for one function, and expect specific, UX friendly exception in such case

Reply:
On it...